src/rulesets/Base/mapreduce.jl

```diff
@@ -48,3 +48,17 @@ function rrule(
     end
     return y, sum_abs2_pullback
 end
+
+####
+#### prod
+####
+
+function rrule(::typeof(prod), x::AbstractArray{T}; dims=:) where {T<:CommutativeMulNumber}
```

Comment:
Would you consider restricting the type of `x`?

Reply:
I just matched `sum`

Reply:
Given that we've not tied this down, I propose `StridedArray`.